lib/engine/game/g_18_ireland/step/merge.rb

```diff
@@ -248,13 +249,39 @@ def mergeable_type(entity)
             end
           end
 
+          def corporation_connected?(c1, c2)
+            # Are corporations connected (blocking by tokens doesn't matter)
+            unless @corp_connectivity[c1]
+              new_home = c1.tokens.first.city
+              connected = []
+              visited = {}
+              new_home.walk(skip_track: :narrow, tile_type: @game.class::TILE_TYPE) do |path, _, _|
```

Comment:
is this expensive? do you think we should add a non blocking graph?

Reply:
in terms of being expensive, it is done in the UI (and when not loading), and it's only done once per each unique network, which is probably as cheap as it can be